app_check/src/common/debug_provider_common.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "firebase/app_check/debug_provider.h"
+
+// Include the header that matches the platform being used.
+#if FIREBASE_PLATFORM_ANDROID
+#include "app_check/src/android/debug_provider_android.h"
+#elif FIREBASE_PLATFORM_IOS || FIREBASE_PLATFORM_TVOS
+#include "app_check/src/ios/debug_provider_ios.h"
+#else
+#include "app_check/src/desktop/debug_provider_desktop.h"
+#endif  // FIREBASE_PLATFORM_ANDROID, FIREBASE_PLATFORM_IOS,
+        // FIREBASE_PLATFORM_TVOS
+
+namespace firebase {
+namespace app_check {
+
+static DebugAppCheckProviderFactory* g_debug_app_check_provider_factory =
+    nullptr;
+
+DebugAppCheckProviderFactory* DebugAppCheckProviderFactory::GetInstance() {
+  if (!g_debug_app_check_provider_factory) {
+    g_debug_app_check_provider_factory = new DebugAppCheckProviderFactory();
+  }
+  return g_debug_app_check_provider_factory;
+}
+
+DebugAppCheckProviderFactory::DebugAppCheckProviderFactory()
+    : internal_(new internal::DebugAppCheckProviderFactoryInternal()) {}
+
+DebugAppCheckProviderFactory::~DebugAppCheckProviderFactory() {
+  if (internal_) {
+    delete internal_;
```

Comment:
What is the significance of setting internal_ to nullptr after delete has already been called?

Reply:
It isn't really needed, but it is just a way to validate that if anything calls into this after the deletion (which ideally shouldn't happen), it will hit a nullptr on internal_ and stop.

Reply:
Alright, seems reasonable.